mmcv/ops/filtered_lrelu.py

```diff
@@ -0,0 +1,435 @@
+# Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
+# and any modifications thereto.  Any use, reproduction, disclosure or
+# distribution of this software and related documentation without an express
+# license agreement from NVIDIA CORPORATION is strictly prohibited.
+
+# source: https://github.com/NVlabs/stylegan3/blob/main/torch_utils/ops/filtered_lrelu.py # noqa
+import warnings
+from typing import Dict, Optional, Union
+
+import numpy as np
+import torch
+
+from ..utils import ext_loader
+from .bias_act import bias_act
+from .upfirdn2d import upfirdn2d
+
+ext_module = ext_loader.load_ext('_ext',
+                                 ['filtered_lrelu', 'filtered_lrelu_act_'])
+
+_plugin = None
+
+
+def _get_filter_size(f):
+    if f is None:
+        return 1, 1
+    assert isinstance(f, torch.Tensor)
+    assert 1 <= f.ndim <= 2
+    return f.shape[-1], f.shape[0]  # width, height
+
+
+def _parse_padding(padding):
+    if isinstance(padding, int):
+        padding = [padding, padding]
+    assert isinstance(padding, (list, tuple))
+    assert all(isinstance(x, (int, np.integer)) for x in padding)
+    padding = [int(x) for x in padding]
+    if len(padding) == 2:
+        px, py = padding
+        padding = [px, px, py, py]
+    px0, px1, py0, py1 = padding
+    return px0, px1, py0, py1
```

Comment:
Those helper functions have been defined at `mmcv/ops/upfirdn2d.py` so we can import them for it.